tests/ui/unsized-locals/extern-parameter-issue-123887.rs

```diff
@@ -0,0 +1,12 @@
+// https://github.com/rust-lang/rust/issues/123887
+// Do not ICE on unsized extern parameter
+//@ compile-flags: -Clink-dead-code --emit=link
+#![feature(extern_types, unsized_fn_params)]
+
+extern "C" {
+    type ExternType;
+}
+
+fn f(_: ExternType) {} //~ ERROR the size for values of type `ExternType` cannot be known at compilation time
```

Comment:
Also, what happens when we monomorphize some `fn f<T: ?Sized>(_: T)` by calling it with a value of `ExternType`?

Reply:
Another good point!  I'm not hugely familiar with the monomorphization code, but have pushed a solution d98f42b1021923d343df7f4568e80f4b5a1ff33f.  I'm not hugely comfortable with it though:

1. it can trigger new recursion depth failures (perhaps this can't be avoided however?);
2. I guess it could end up being evaluated for the same mono item multiple times, which would be unnecessary—but I don't quite understand the monomorphization process well enough to see how this can best be avoided?
3. does it need a perf run?